File: app/src/main/java/pl/piotrskiba/dailywallpaper/asynctasks/SetWallpaperAsyncTask.java
package pl.piotrskiba.dailywallpaper.asynctasks;

import android.app.WallpaperManager;
import android.content.Context;
import android.graphics.Bitmap;
import android.os.AsyncTask;

import java.io.IOException;

import pl.piotrskiba.dailywallpaper.interfaces.WallpaperSetListener;

public class SetWallpaperAsyncTask extends AsyncTask<Bitmap, Void, Boolean> {

    private final Context context;
    private final WallpaperSetListener listener;

    public SetWallpaperAsyncTask(Context context, WallpaperSetListener listener){
        this.context = context;
        this.listener = listener;
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
    }

    @Override
    protected Boolean doInBackground(Bitmap... bitmaps) {
        WallpaperManager wallpaperManager = WallpaperManager.getInstance(context);

        try {
            wallpaperManager.setBitmap(bitmaps[0]);
            return true;
        } catch (IOException e) {
            e.printStackTrace();
        }

        return false;
    }

    @Override
    protected void onPostExecute(Boolean aBoolean) {
        listener.onWallpaperSet(aBoolean);
        super.onPostExecute(aBoolean);
    }
}

File: app/src/main/java/pl/piotrskiba/dailywallpaper/asynctasks/SetWallpaperAsyncTask.kt
package pl.piotrskiba.dailywallpaper.asynctasks

import android.app.WallpaperManager
import android.content.Context
import android.graphics.Bitmap
import android.os.AsyncTask
import pl.piotrskiba.dailywallpaper.interfaces.WallpaperSetListener
import java.io.IOException

class SetWallpaperAsyncTask(private val context: Context, private val listener: WallpaperSetListener) : AsyncTask<Bitmap, Unit, Boolean>() {
    override fun onPreExecute() {
        super.onPreExecute()
    }

    override fun doInBackground(vararg bitmaps: Bitmap): Boolean {
        val wallpaperManager = WallpaperManager.getInstance(context)
        try {
            wallpaperManager.setBitmap(bitmaps[0])
            return true
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return false
    }

    override fun onPostExecute(aBoolean: Boolean) {
        listener.onWallpaperSet(aBoolean)
        super.onPostExecute(aBoolean)
    }

}